### DISCUSSION TITLE
[Question] How to trigger route change event on page initialization?

### DISCUSSION THREAD

**Discussion by seriaati** [flet-dev/flet#3921] [Q&A]
### Question

I have a router system set up with `page.on_route_change`. However, when the page is just loaded, it doesn't trigger the route change event. For example, if I have the `on_route_change` function set up so that it would add a view when the route is `/login`, and the user straight up goes to `localhost:8080/login` (where the flet application is hosted at), the view won't be added. I want it so that when the page is just initialized, it would also go through the router logic.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by factor3** [comment]
Please note: I am having a similar problem. Apparently, when the flet application starts, `it does not call the on_change_route function that I set. The difference between my problem and @seriaati 's problem is that I am calling page.push_root('/') instead of a go('login'). Has anyone actually answered this?

**Reply by seriaati** [reply]
Hi, this was from 2 years ago and I genuinely forgot how I solved it. Looking at my current source code in https://github.com/seriaati/hoyo-buddy/blob/d79e5f255a9fb6eae5d34ffad666773cbe778d9e/hoyo_buddy/web_app/app.py#L74 , it seems like I am calling `page.go(page.route)` after setting the custom router. I hope this helps.
